templates/boltzmann/core/honeycomb.ts

```diff
@@ -0,0 +1,909 @@
+    )
+
+    this.initialized = true
+  }
+
+  // Start the OpenTelemetry SDK. If using beelines, this is
+  // a no-op. This needs to happen before anything happens in
+  // the entrypoint and is an async operation.
+  public async start(): Promise<void> {
+    let exitCode = 0
+    const sdk = this.sdk
+
+    async function die(err: Error) {
+      console.error(err.stack)
+      exitCode = 1
+      await shutdown()
+    }
+
+    async function shutdown() {
+      if (sdk) {
+        try {
+          await sdk.shutdown()
+        } catch (err) {
+          console.error(err.stack)
+        }
+      }
+      process.exit(exitCode)
+    }
+
+    if (sdk) {
+      process.once('SIGTERM', shutdown)
+      process.once('beforeExit', shutdown)
+      process.once('uncaughtException', die)
+      process.once('unhandledRejection', die)
+      await sdk.start()
+      this.started = true
+    }
+  }
+
+  // Create a trace, call the runInContext function, then close the
+  // trace after it resolved. For beeline, this is is a Trace;
+  // for OpenTelemetry, it's a root span.
+  public async withTrace(
+    context: Context,
+    runInContext: () => Promise<any>,
+    headerSources?: string[]
+  ): Promise<any> {
+    if (!this.features.honeycomb) {
+      // Call legacy beelines implementation
+      return this.withBeelineTrace(context, runInContext, headerSources)
+    }
+
+    if (headerSources) {
+      console.warn('trace headerSources are a beeline-only feature')
+    }
+
+    /*
+     * ┏┓
+     * ┃┃╱╲ in
+     * ┃╱╱╲╲ this
+     * ╱╱╭╮╲╲house
+     * ▔▏┗┛▕▔ we
+     * ╱▔▔▔▔▔▔▔▔▔▔╲
+     * trace with opentelemetry
+     * ╱╱┏┳┓╭╮┏┳┓ ╲╲
+     * ▔▏┗┻┛┃┃┗┻┛▕▔
+     */
+
+    const tracer = this.tracer
+    let carrier = {}
+
+    // Start a parent span
+    const parentSpan = tracer.startSpan(
+      `${context.method} ${context.url.pathname}${context.url.search}`,
+      {
+        attributes: {
+          [SemanticAttributes.HTTP_HOST]: context.host,
+          [SemanticAttributes.HTTP_URL]: context.url.href,
+          [SemanticAttributes.NET_PEER_IP]: context.remote,
+          [SemanticAttributes.HTTP_METHOD]: context.method,
+          [SemanticAttributes.HTTP_SCHEME]: context.url.protocol,
+          [SemanticAttributes.HTTP_ROUTE]: context.url.pathname,
+          // TODO: Honeycomb.Attributes obvs
+          [Honeycomb.OTEL_REQ_QUERY]: context.url.search
+        }
+      }
+    )
+
+    // this propagator takes care of extracting trace parent
+    // and state from request headers (and so on)
+    const propagator = new W3CTraceContextPropagator()
+
+    propagator.inject(
+      trace.setSpanContext(
+        ROOT_CONTEXT,
+        parentSpan.spanContext()
+      ),
+      carrier,
+      defaultTextMapSetter
+    )
+
+    /* TODO: Do I need to create and set a context? No, right?
+
+    // create a parent active context
+    const parentContext = propagator.extract(
+      ROOT_CONTEXT,
+      carrier,
+      defaultTextMapGetter
+    )
```

Comment:
this is backwards in terms of what's commented out :)
I think I spell it all out in the last round of comments that I left.
Do you want to go through those together on video sometime or something?

Reply:
I want to review your comments more closely and see if I can follow them first - you gave me a lot of good direction and I just haven't followed up on it - but let's do a review on zoom when I get there. I think it'll be helpful!

Reply:
ok cool. Yeah I think the round of comments from a few days ago (plus the couple since then) should make get things in the right shape for testing. https://github.com/entropic-dev/boltzmann/pull/101#pullrequestreview-861541439

Let me know when you want to go over things on zoom, send a meeting invite anytime!!